core/src/main/java/org/axonframework/queryhandling/backpressure/ThrottleFirstBackpressure.java

```diff
@@ -0,0 +1,78 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ *
+ */
+
+package org.axonframework.queryhandling.backpressure;
+
+import org.axonframework.queryhandling.UpdateHandler;
+
+import java.util.concurrent.Executors;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.AtomicReference;
+
+/**
+ * Throttle First backpressure mechanism - the {@code original} {@link UpdateHandler} will be invoked after given
+ * period of time with first update received. If there were no updates within given period, {@code original} will not be
+ * invoked. Do note that invocation of {@code original} will be done in separate (worker) thread.
+ *
+ * @param <I> type of initial result
+ * @param <U> type of incremental updates
+ * @author Milan Savic
+ * @since 3.3
+ */
+public class ThrottleFirstBackpressure<I, U> extends TimeBasedBackpressure<I, U> {
+
+    private final AtomicReference<U> firstUpdate = new AtomicReference<>();
+
+    /**
+     * Initializes {@link ThrottleFirstBackpressure} with original update handler and parameters for scheduling.
+     *
+     * @param original     the original update handler
+     * @param initialDelay the delay after which to start scheduling of updates
+     * @param period       the period on which to schedule updates
+     * @param unit         time unit
+     */
+    public ThrottleFirstBackpressure(UpdateHandler<I, U> original, long initialDelay, long period, TimeUnit unit) {
+        this(original, initialDelay, period, unit, Executors.newSingleThreadScheduledExecutor());
+    }
+
+    /**
+     * Initializes {@link ThrottleFirstBackpressure} with original update handler and parameters for scheduling.
+     *
+     * @param original                 the original update handler
+     * @param initialDelay             the delay after which to start scheduling of updates
+     * @param period                   the period on which to schedule updates
+     * @param unit                     time unit
+     * @param scheduledExecutorService scheduled executor service
+     */
+    public ThrottleFirstBackpressure(UpdateHandler<I, U> original, long initialDelay, long period, TimeUnit unit,
+                                     ScheduledExecutorService scheduledExecutorService) {
+        super(original, period, unit, scheduledExecutorService);
+        scheduledExecutorService.scheduleAtFixedRate(() -> {
+            U current = firstUpdate.getAndSet(null);
+            if (current != null) {
+                original.onUpdate(current);
+            }
+        }, initialDelay, period, unit);
+    }
+
+    @Override
+    public void onUpdate(U update) {
+        firstUpdate.compareAndSet(null, update);
```

Comment:
Wouldn't this call return `false` if the expected `null` isn't true? So, wouldn't this `onUpdate` call fail if there is already a scheduled update which hasn't been done yet?
Not sure whether that's desirable :thinking:

Reply:
The idea with throttle first backpressure mechanism is to have only first update propagated to the `delegateUpdateHandler` within given period of time. So, we want to change the value of first update only if it is first within given time frame - to us, this means that we have set it to null.

Reply:
I would suggest combining the two backpressure class (first and last), and use a BiFunction<U, U, U> that defines what to actually send. In that case, someone can choose to return the first, the last or even a combination of the two. One might, for example, combine two received lists into a single one.

Reply:
Makes sense. I would introduce `ThrottleBackpressure` class doing exactly what @abuijze described, but would leave `ThrottleFirstBackpressure` and `SampleBackpressure` (which would inherit `ThrottleBackpressure` by specifying BiFunction to pass first/last update respectively) for convenience. Sounds good?